test/plugins/CMakeLists.txt

```diff
@@ -31,7 +31,7 @@ endif()
 # Plugin shared libraries
 if(BUILD_TESTING)
   foreach (src ${test_plugins})
-    add_library(${src} SHARED ${src}.cc)
+    add_library(${src} MODULE ${src}.cc)
```

Comment:
I think this has caused a windows test failure

* https://build.osrfoundation.org/job/gz_sim-pr-cnlwin/45/testReport/junit/(root)/SystemLoader/FromPluginPathEnv/

I think we should be able to update the test to account for the change

Reply:
Done in [fb73007](https://github.com/gazebosim/gz-sim/pull/2849/commits/fb73007d24ba03e74d51568884ea454e8a11e518)